packages/components/src/components/Prompt/index.native.tsx

```diff
@@ -1,88 +1,119 @@
 import React from 'react';
 import styled from 'styled-components/native';
-import PropTypes from 'prop-types';
+import { Animated, Easing } from 'react-native';
 import Icon from '../Icon';
 
 import icons from '../../config/icons';
 import { Omit, IconShape } from '../../support/types';
 import colors from '../../config/colors';
 
-// TODO: Rewrite using Animated API
-// const PulseAnimation = keyframes`
-//     0% {
-//         background-color: ${colors.GREEN_PRIMARY};
-//         transform: scale(0);
-//         opacity: 0;
-//     }
-//     25% {
-//         background-color: ${colors.GREEN_PRIMARY};
-//         transform: scale(0.75);
-//         opacity: 0.2;
-//     }
-//     50% {
-//         transform: scale(1.5);
-//         opacity: 0.3;
-//     }
-//     100% {
-//         opacity: 0;
-//         transform: scale(4);
-//     }
-// `;
+const Pulse = styled.View<Omit<Props, 'model'>>`
+    background-color: ${colors.GREEN_PRIMARY};
+    opacity: 0.3;
+    border-radius: 100;
+    height: ${props => props.size};
+    width: ${props => props.size};
+    margin-left: -${props => ((props.size || 32) - (props.size || 32) * (props.ratio || 1)) / 2}px;
+`;
 
 const IconWrapper = styled.View<Omit<Props, 'model'>>`
-    position: relative;
-    height: ${props => props.size}px;
-    width: ${props => props.size}px;
+    width: ${props => (props.size || 32) * (props.ratio || 1)};
+    height: ${props => props.size};
 `;
 
 const Wrapper = styled.View`
-    display: flex;
+    flex: 1;
     flex-direction: column;
     justify-content: center;
     align-items: center;
     text-align: center;
 `;
 
-const ContentWrapper = styled.View`
-    max-width: 300px;
+const ContentWrapper = styled.Text`
     color: ${colors.GREEN_PRIMARY};
     text-align: center;
-    margin: 5px;
+    margin: 10px;
+`;
+
+const Animation = styled(Animated.View)`
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
 `;
 
-const modelToIcon = (model: number) => {
+const modelToIcon = (model: model) => {
     const mapping: { [key: number]: IconShape } = {
         1: icons.T1,
         2: icons.T2,
     };
     return mapping[model];
 };
+
+type model = 1 | 2;
+
 interface Props {
-    model: number;
+    model: model;
     size?: number;
+    ratio?: number;
     children?: React.ReactNode;
 }
 
-const Prompt = ({ model, size, children }: Props) => {
-    return (
-        <Wrapper>
-            <IconWrapper size={size}>
-                {/* <Pulse /> */}
-                <Icon icon={modelToIcon(model)} size={size} color={colors.GREEN_PRIMARY} />
-            </IconWrapper>
-            <ContentWrapper>{children}</ContentWrapper>
-        </Wrapper>
-    );
-};
+class Prompt extends React.Component<Props> {
+    state = {
+        blinkAnim: new Animated.Value(0),
+    };
 
-Prompt.propTypes = {
```

Comment:
propTypes missing (I think we should keep them as they are still useful in projects that don't use typescript)

Reply:
It has been fixed in this commit https://github.com/trezor/trezor-suite/commit/57f3af200a8874aaf1a258f45c6dea96c889a539#diff-321317b7701314169a54ad71b261a8b4R69